examples/network/hierarchy_network_demo.py

```diff
@@ -0,0 +1,358 @@
+             "maxSegmentsPerCell": 128,
+             "initialPerm": 0.21,
+             "permanenceInc": 0.1,
+             "permanenceDec": 0.1,
+             "globalDecay": 0.0,
+             "maxAge": 0,
+             "minThreshold": 9,
+             "activationThreshold": 12,
+             "outputType": "normal",
+             "pamLength": 3}
+
+_RECORD_SENSOR = "sensorRegion"
+_L1_SPATIAL_POOLER = "l1SpatialPoolerRegion"
+_L1_TEMPORAL_POOLER = "l1TemporalPoolerRegion"
+_L1_CLASSIFIER = "l1Classifier"
+
+_L2_SPATIAL_POOLER = "l2SpatialPoolerRegion"
+_L2_TEMPORAL_POOLER = "l2TemporalPoolerRegion"
+_L2_CLASSIFIER = "l2Classifier"
+
+
+def createEncoder():
+  """
+  Creates and returns a #MultiEncoder including a ScalarEncoder for
+  energy consumption and a DateEncoder for the time of the day.
+
+  @see nupic/encoders/__init__.py for type to file-name mapping
+  @see nupic/encoders for encoder source files
+  """
+  encoder = MultiEncoder()
+  encoder.addMultipleEncoders({
+      "consumption": {"fieldname": u"consumption",
+                      "type": "ScalarEncoder",
+                      "name": u"consumption",
+                      "minval": 0.0,
+                      "maxval": 100.0,
+                      "clipInput": True,
+                      "w": 21,
+                      "n": 500},
+      "timestamp_timeOfDay": {"fieldname": u"timestamp",
+                              "type": "DateEncoder",
+                              "name": u"timestamp_timeOfDay",
+                              "timeOfDay": (21, 9.5)}
+  })
+  return encoder
+
+
+def createRecordSensor(network, name, dataSource):
+  """
+  Creates a RecordSensor region that allows us to specify a file record
+  stream as the input source.
+  """
+
+  # Specific type of region. Possible options can be found in /nupic/regions/
+  regionType = "py.RecordSensor"
+
+  # Creates a json from specified dictionary.
+  regionParams = json.dumps({"verbosity": _VERBOSITY})
+  network.addRegion(name, regionType, regionParams)
+
+  # getSelf returns the actual region, instead of a region wrapper
+  sensorRegion = network.regions[name].getSelf()
+
+  # Specify how RecordSensor encodes input values
+  sensorRegion.encoder = createEncoder()
+
+  # Specify the dataSource as a file record stream instance
+  sensorRegion.dataSource = dataSource
+  return sensorRegion
+
+
+def createSpatialPooler(network, name, inputWidth):
+  # Create the spatial pooler region
+  SP_PARAMS["inputWidth"] = inputWidth
+  spatialPoolerRegion = network.addRegion(name, "py.SPRegion",
+                                          json.dumps(SP_PARAMS))
+  # Make sure learning is enabled
+  spatialPoolerRegion.setParameter("learningMode", True)
+  # We want temporal anomalies so disable anomalyMode in the SP. This mode is
+  # used for computing anomalies in a non-temporal model.
+  spatialPoolerRegion.setParameter("anomalyMode", False)
+  return spatialPoolerRegion
+
+
+def createTemporalPooler(network, name):
```

Comment:
I know the region is called TPRegion but please use "temporal memory" to refer to or name variables. We are moving away from calling it a temporal pooler since it isn't doing temporal pooling.